webapp/channels/src/components/announcement_bar/version_bar/version_bar.tsx

```diff
@@ -12,67 +12,52 @@ interface Props {
     buildHash?: string;
 }
 
-interface State {
-    buildHashOnAppLoad?: string;
-}
-
-export default class VersionBar extends React.PureComponent <Props, State> {
-    constructor(props: Props) {
-        super(props);
+const VersionBar = ({
+    buildHash,
+}: Props) => {
+    const [buildHashOnAppLoad, setBuildHashOnAppLoad] = useState(buildHash);
 
-        this.state = {
-            buildHashOnAppLoad: props.buildHash,
-        };
-    }
+    const reloadPage = () => {
+        window.location.reload();
+    };
 
-    static getDerivedStateFromProps(props: Props, state: State) {
-        if (!state.buildHashOnAppLoad && props.buildHash) {
-            return {
-                buildHashOnAppLoad: props.buildHash,
-            };
+    useEffect(() => {
+        if (!buildHashOnAppLoad && buildHash) {
+            setBuildHashOnAppLoad(buildHash);
         }
+    }, [buildHash, buildHashOnAppLoad]);
 
+    if (!buildHashOnAppLoad) {
         return null;
     }
 
-    reloadPage = () => {
-        window.location.reload();
-    };
-
-    render() {
-        const {buildHashOnAppLoad} = this.state;
-        const {buildHash} = this.props;
-
-        if (!buildHashOnAppLoad) {
-            return null;
-        }
-
-        if (buildHashOnAppLoad !== buildHash) {
-            return (
-                <AnnouncementBar
-                    type={AnnouncementBarTypes.ANNOUNCEMENT}
-                    message={
-                        <React.Fragment>
+    if (buildHashOnAppLoad !== buildHash) {
```

Comment:
Since I am asking for changes...

Let's move this the other way around and return early, to avoid one level of indentation. Instead of...
```
if (buildHashOnAppLoad !== buildHash) {
    return (<AnnouncementBar .../>);
}
return null;
```

Lets do:
```
if (buildHashOnAppLoad === buildHash) {
    return null;
}

return (<AnnouncementBar ... />);
```